src/snowflake/snowpark/functions.py

```diff
@@ -8234,7 +8278,16 @@ def table_function(function_name: str) -> Callable:
         >>> session.table_function(split_to_table(lit("split words to table"), lit(" ")).over()).collect()
         [Row(SEQ=1, INDEX=1, VALUE='split'), Row(SEQ=1, INDEX=2, VALUE='words'), Row(SEQ=1, INDEX=3, VALUE='to'), Row(SEQ=1, INDEX=4, VALUE='table')]
     """
-    return lambda *args, **kwargs: call_table_function(function_name, *args, **kwargs)
+    fn = lambda *args, **kwargs: call_table_function(  # noqa: E731
```

Comment:
there is no comment in the original code,  but maybe we could add one to explain why we need to wrap yet in another lambda?

Reply:
(alternative was functools.partial here)

Reply:
I'm not really sure about the design of `functions.table_function`. It just seems like a way of making a Python callable for a function name, e.g.

```python
foo = table_function('foo')
foo(lit(1), lit('bar'))
```

The public API has other inconsistencies and confusing choices, as I mentioned in Slack, too. For example, the signature of `Session.table_function` is `func_name, *args, **kwargs`, but `func_name` can also be this callable, or a `TableFunctionCall`. `TableFunctionCall` in particular takes parameters.

```python
make_table = call_table_function('make_table', lit(42))
```

The meaning of these two is unambiguous:

```python
Session.table_function(make_table)
Session.table_function('make_table', lit(42))
```

What does this do?

```python
Session.table_function(make_table, lit(99))
```